syncmaster/worker/handlers/db/base.py

```diff
@@ -44,10 +45,16 @@ def read(self) -> DataFrame:
         return reader.run()
 
     def write(self, df: DataFrame) -> None:
-        writer = DBWriter(
-            connection=self.connection,
-            table=self.transfer_dto.table_name,
-        )
+        writer_params = {
+            "connection": self.connection,
+            "table": self.transfer_dto.table_name,
+            "options": {"if_exists": "replace_entire_table"},
+        }
+
+        if self.transfer_dto.strategy.type == "incremental" and self.hwm:
+            writer_params["options"]["if_exists"] = "append"
+
+        writer = DBWriter(**writer_params)
```

Comment:
Let's use `DBWriter(connection=..., table=...)` as before, and parametrize only `options` value. It's hard to read